### PR TITLE
Fix nullable annotations for `Preferences`

### PR DESCRIPTION
**Pull request by maxkoshevoi**
### Description of Change

Setting and Getting `null` is totally valid (I'm using this functionality in my app), so existing annotations are incorrect.
Updated annotations to reflect underlying logic.

### Issues Fixed

Fixes #
~~Do I need to create an issue for such a small change?~~

### CODE REVIEW

**Review by rmarinho — DISMISSED** [review]
Build failing 

```
   "D:\a\_work\1\s\Microsoft.Maui.BuildTasks.slnf" (Build target) (1:2) ->
       "D:\a\_work\1\s\src\Core\src\Core.csproj" (default target) (2:40) ->
       "D:\a\_work\1\s\src\Core\src\Core.csproj" (Build target) (2:43) ->
       "D:\a\_work\1\s\src\Essentials\src\Essentials.csproj" (default target) (8:55) ->
       (CoreCompile target) -> 
         D:\a\_work\1\s\src\Essentials\src\Preferences\Preferences.shared.cs(38,12): error CS0246: The type or namespace name 'NotNullIfNotNullAttribute' could not be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Essentials\src\Essentials.csproj]
         D:\a\_work\1\s\src\Essentials\src\Preferences\Preferences.shared.cs(38,12): error CS0246: The type or namespace name 'NotNullIfNotNull' could not be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Essentials\src\Essentials.csproj]
         D:\a\_work\1\s\src\Essentials\src\Preferences\Preferences.shared.cs(101,12): error CS0246: The type or namespace name 'NotNullIfNotNullAttribute' could not be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Essentials\src\Essentials.csproj]
         D:\a\_work\1\s\src\Essentials\src\Preferences\Preferences.shared.cs(101,12): error CS0246: The type or namespace name 'NotNullIfNotNull' could not be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Essentials\src\Essentials.csproj]
```

**Review by mattleibow — APPROVED** [review]

File: Directory.Build.props [hunk 1]
```diff
@@ -87,6 +87,9 @@
     <PackageVersion>$(DotNetVersionBand)-dev</PackageVersion>
     <PackageOutputPath>$(MSBuildThisFileDirectory)artifacts</PackageOutputPath>
   </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Nullable" Version="1.3.0" PrivateAssets="all" />
+  </ItemGroup>
```
Comment (mattleibow) [inline Directory.Build.props:92]: This is causing a whole bunch of extra errors. I wonder if either we skip the nullable attributes on netstandard or we only include this on netstandard?

**Review by mattleibow — CHANGES_REQUESTED** [review]

File: Directory.Build.props (hunk 1 above)
Comment (maxkoshevoi) [reply Directory.Build.props]: What were the errors? I couldn't figure them out from the CI

Added condition to only include it for netstandard2.0

**Review by mattleibow — APPROVED** [review]

File: Directory.Build.props [hunk 2]
```diff
@@ -87,6 +87,9 @@
     <PackageVersion>$(DotNetVersionBand)-dev</PackageVersion>
     <PackageOutputPath>$(MSBuildThisFileDirectory)artifacts</PackageOutputPath>
   </PropertyGroup>
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
+    <PackageReference Include="Nullable" Version="1.3.0" PrivateAssets="all" />
+  </ItemGroup>
```
Comment (mattleibow) [inline Directory.Build.props:92]: This package is causing a bunch of issues as it is adding a public type to all assemblies it seems: 
https://dev.azure.com/xamarin/public/_build/results?buildId=60634&view=logs&j=a2b1c795-2687-5fd3-b418-4e4b87c7d580&t=f98b3571-5084-5d4e-0013-9a861ec79559

File: src/Essentials/src/Preferences/Preferences.shared.cs [hunk 3]
```diff
@@ -34,7 +35,8 @@ public static class Preferences
 			Clear(null);
 
 		/// <include file="../../docs/Microsoft.Maui.Essentials/Preferences.xml" path="//Member[@MemberName='Get'][7]/Docs" />
-		public static string Get(string key, string defaultValue) =>
+		[return: NotNullIfNotNull("defaultValue")]
```
Comment (mattleibow) [inline src/Essentials/src/Preferences/Preferences.shared.cs:38]:
```suggestion
#if !NETSTANDARD2_0 && !NETSTANDARD2_1
		[return: NotNullIfNotNull("defaultValue")]
#endif
```

File: src/Essentials/src/Preferences/Preferences.shared.cs [hunk 4]
```diff
@@ -96,8 +98,9 @@ public static class Preferences
 			Current.Clear(sharedName);
 
 		/// <include file="../../docs/Microsoft.Maui.Essentials/Preferences.xml" path="//Member[@MemberName='Get'][14]/Docs" />
-		public static string Get(string key, string defaultValue, string? sharedName) =>
-			Current.Get<string>(key, defaultValue, sharedName);
+		[return: NotNullIfNotNull("defaultValue")]
```
Comment (mattleibow) [inline src/Essentials/src/Preferences/Preferences.shared.cs:101]:
```suggestion
#if !NETSTANDARD2_0 && !NETSTANDARD2_1
		[return: NotNullIfNotNull("defaultValue")]
#endif
```

**Review by mattleibow — CHANGES_REQUESTED** [review]
The new nuget is causing issues and probably not worth it at all. netstandard is basically just there for the IDE tools, and they don't need all these things. Simplest is just to condition it out for the obsolete platforms.

**Review by mattleibow — APPROVED** [review]

**Pull request merged**